src/network/mod.rs

```diff
@@ -0,0 +1,116 @@
+//! Network types and implementations.
+
+use crate::sandbox::SandboxData;
+use anyhow::Result;
+use derive_builder::Builder;
+
+pub mod cni;
+
+#[derive(Builder)]
+#[builder(pattern = "owned", setter(into))]
+/// Network is the main structure for working with the Container Network Interface.
+/// The implementation `T` can vary and is being defined in the `Pod` trait.
+pub struct Network<T>
+where
+    T: Default,
+{
+    #[builder(default = "T::default()")]
+    /// Trait implementation for the network.
+    implementation: T,
+}
+
+/// Common network behavior trait
+pub trait PodNetwork {
+    /// Initialize the network, which will be usually done once on application startup.
+    fn initialize(&mut self) -> Result<()> {
```

Comment:
Do we need an initialize method?

Reply:
The idea was to register the file-watchers here for example. This happens only once during server startup and seems to be a good fit. :) Any other thoughts on that?